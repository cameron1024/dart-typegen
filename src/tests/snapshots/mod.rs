// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
