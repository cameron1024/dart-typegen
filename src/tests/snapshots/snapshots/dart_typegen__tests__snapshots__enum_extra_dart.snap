---
source: src/tests/snapshots/mod.rs
expression: dart
---
enum Foo {
  a,
  b;

  factory Foo.fromJson(dynamic json) => switch (json) {
    "a" => Foo.a,
    "b" => Foo.b,
    final other => throw ArgumentError("Unknown variant: $other"),
  };

  dynamic toJson() => switch (this) {
    Foo.a => "a",
    Foo.b => "b",
  };
  @override
  String toString() => switch (this) {
    Foo.a => "a",
    Foo.b => "b",
  };

  void foo() => print(this);
}
